server/src/test/java/org/elasticsearch/transport/Lz4TransportDecompressorTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.transport;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.bytes.CompositeBytesReference;
+import org.elasticsearch.common.bytes.ReleasableBytesReference;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.OutputStreamStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.util.PageCacheRecycler;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class Lz4TransportDecompressorTests extends ESTestCase {
+
+    public void testSimpleCompression() throws IOException {
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            byte randomByte = randomByte();
+            try (OutputStream lz4BlockStream = Compression.Scheme.lz4OutputStream(Streams.noCloseStream(output))) {
+                lz4BlockStream.write(randomByte);
+            }
+
+            BytesReference bytes = output.bytes();
+
+            Lz4TransportDecompressor decompressor = new Lz4TransportDecompressor(PageCacheRecycler.NON_RECYCLING_INSTANCE);
+            int bytesConsumed = decompressor.decompress(bytes);
+            assertEquals(bytes.length(), bytesConsumed);
+            ReleasableBytesReference releasableBytesReference = decompressor.pollDecompressedPage(true);
+            assertEquals(randomByte, releasableBytesReference.get(0));
+            releasableBytesReference.close();
+
+        }
+    }
+
+    public void testMultiPageCompression() throws IOException {
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            try (StreamOutput lz4BlockStream = new OutputStreamStreamOutput(Compression.Scheme.lz4OutputStream(
+                Streams.flushOnCloseStream(output)))) {
+                for (int i = 0; i < 10000; ++i) {
+                    lz4BlockStream.writeInt(i);
```

Comment:
Can we write randomInt's instead? Or perhaps randomly also write just the same random int for all entries to allow max compression?